commit 8b758859dfbe9598ba41e8b9b01e44edcc0c2fc1
Author: Namjae Jeon <namjae.jeon@samsung.com>
Date:   Tue Jun 29 14:52:00 2021 +0900

    ksmbd: reorder and document on-disk and netlink structures in headers
    
    Reorder and document on-disk and netlink structures in headers.
    
    This is a userspace ABI to communicate data between ksmbd and user IPC
    daemon using netlink. This is added to track and cache user account DB
    and share configuration info from userspace.
    
     - KSMBD_EVENT_HEARTBEAT_REQUEST(ksmbd_heartbeat)
       This event is to check whether user IPC daemon is alive. If user IPC
       daemon is dead, ksmbd keep existing connection till disconnecting and
       new connection will be denied.
    
     - KSMBD_EVENT_STARTING_UP(ksmbd_startup_request)
       This event is to receive the information that initializes the ksmbd
       server from the user IPC daemon and to start the server. The global
       section parameters are given from smb.conf as initialization
       information.
    
     - KSMBD_EVENT_SHUTTING_DOWN(ksmbd_shutdown_request)
       This event is to shutdown ksmbd server.
    
     - KSMBD_EVENT_LOGIN_REQUEST/RESPONSE(ksmbd_login_request/response)
       This event is to get user account info to user IPC daemon.
    
     - KSMBD_EVENT_SHARE_CONFIG_REQUEST/RESPONSE
       (ksmbd_share_config_request/response)
       This event is to get net share configuration info.
    
     - KSMBD_EVENT_TREE_CONNECT_REQUEST/RESPONSE
       (ksmbd_tree_connect_request/response)
       This event is to get session and tree connect info.
    
     - KSMBD_EVENT_TREE_DISCONNECT_REQUEST(ksmbd_tree_disconnect_request)
       This event is to send tree disconnect info to user IPC daemon.
    
     - KSMBD_EVENT_LOGOUT_REQUEST(ksmbd_logout_request)
       This event is to send logout request to user IPC daemon.
    
     - KSMBD_EVENT_RPC_REQUEST/RESPONSE(ksmbd_rpc_command)
       This event is to make DCE/RPC request like srvsvc, wkssvc, lsarpc,
       samr to be processed in userspace.
    
     - KSMBD_EVENT_SPNEGO_AUTHEN_REQUEST/RESPONSE
       (ksmbd_spnego_authen_request/response)
       This event is to make kerberos authentication to be processed in
       userspace.
    
    Reviewed-by: Christoph Hellwig <hch@lst.de>
    Signed-off-by: Namjae Jeon <namjae.jeon@samsung.com>
    Signed-off-by: Steve French <stfrench@microsoft.com>

diff --git a/fs/ksmbd/mgmt/tree_connect.h b/fs/ksmbd/mgmt/tree_connect.h
index 4e40ec3f4774..18e2a996e0aa 100644
--- a/fs/ksmbd/mgmt/tree_connect.h
+++ b/fs/ksmbd/mgmt/tree_connect.h
@@ -8,7 +8,7 @@
 
 #include <linux/hashtable.h>
 
-#include "../ksmbd_server.h"
+#include "../ksmbd_netlink.h"
 
 struct ksmbd_share_config;
 struct ksmbd_user;